parser.cpp

```diff
@@ -0,0 +1,88 @@
+#include "parser.h"
+
+std::map<std::string, std::string> Parser::ParseJsonString(std::string StringToParse){
+    std::map<std::string, std::string> Map;
+    bool toDelete = true;
+    int j = 0;
+    while (j < StringToParse.length())
+    {
+        if (StringToParse[j] == '"'){
+            if (toDelete)
+            {
+                toDelete = false;
+            }else{
+                toDelete = true;
+            }
+        }
+        if (StringToParse[j] == ' ' && toDelete)
+        {
+            StringToParse.erase(j,1);
+        }else{
+            j++;
+        }
+    }
+    std::string Str = StringToParse;
+    std::string name = "name";
+    std::string hp = "hp";
+    std::string dmg = "dmg";
+    
+    Map["name"] = Parser::FindData(Str,name);
+
+    Map["hp"] = Parser::FindData(Str,hp);
+
+    Map["dmg"] = Parser::FindData(Str,dmg);
+
+    return Map;
+}
+
+std::map<std::string, std::string> Parser::ParseJson(std::istream& JSONToParse){
+    std::string DataToParse, line;
+    while (std::getline(JSONToParse, line))
+    {
+        DataToParse += line;
+    }
+    
+    return Parser::ParseJsonString(DataToParse);
+}
+
+std::map<std::string, std::string> Parser::ParseJsonFilename(std::string FilenameToParse){
+    std::ifstream ToParse(FilenameToParse);
+    std::string DataToParse = "";
+    if (ToParse.good())
+    {
+        std::string line;
+        while (std::getline(ToParse, line))
+        {
+            DataToParse += line;
+        }
+    }else{
+        
+    }
+    return Parser::ParseJsonString(DataToParse);
+}
+
+std::string Parser::FindData(const std::string& StringToParse, const std::string& StringToFind){
+    std::string data = "";
+
+    if (StringToParse.find(StringToFind) != std::string::npos)
+    {
+        int findWord = StringToParse.find(StringToFind) + StringToFind.length() + 2;
+         if (StringToParse[findWord] == '"')
+        {
+            findWord++;
+        }
+        do
+        {
+            data += StringToParse[findWord];
+            findWord++;
+        } while (isdigit(StringToParse[findWord]) or isalpha(StringToParse[findWord]));
```

Comment:
Akkor a szóközöket pl. nem másoljuk? És ha rögtön az első karakter sem megfelelő?

Reply:
Köszönöm az észrevételt, belevettem a szóközöket is, véletlenül maradtak ki.
Ha nincs megfelelő karakter a data string üres marad amire később dobunk egy runtime errort!